slasher/db/iface/interface.go

```diff
@@ -94,11 +96,16 @@ type FullAccessDatabase interface {
 	WriteAccessDatabase
 }
 
+// EpochSpans represents a data access layer for marshaling and unmarshaling validator spans for each validator per epoch.
+type EpochSpans interface {
+	SetValidatorSpan(ctx context.Context, idx uint64, newSpan detectionTypes.Span) error
+	GetValidatorSpan(ctx context.Context, idx uint64) (detectionTypes.Span, error)
+}
```

Comment:
Maybe this should go in `slasher/types`?

Reply:
i think its good here. maybe another db impl will need to move it to db somehow